src/Security/RolesDownloadStrategy.php

```diff
@@ -35,16 +36,32 @@ class RolesDownloadStrategy implements DownloadStrategyInterface
     protected $security;
 
     /**
-     * @var TranslatorInterface
+     * @var LegacyTranslatorInterface|TranslatorInterface
      */
     protected $translator;
 
     /**
      * @param AuthorizationCheckerInterface $security
      * @param string[]                      $roles
      */
-    public function __construct(TranslatorInterface $translator, $security, array $roles = [])
+    public function __construct(object $translator, $security, array $roles = [])
     {
+        if (!$translator instanceof TranslatorInterface) {
+            @trigger_error(sprintf(
+                'Argument 1 should be an instance of "%s" and will be required in 4.0.',
+                TranslatorInterface::class
```

Comment:
The argument 1 is already required, since it hasn't a default value.
```suggestion
        if (!$translator instanceof TranslatorInterface) {
            @trigger_error(sprintf(
                'Argument 1 passed to "%s()" MUST be an instance of "%s", "%s" given.',
                __METHOD__,
                TranslatorInterface::class,
                \get_class($translator)
```